pkg/security/ImageScanService.go

```diff
@@ -365,15 +381,31 @@ func (impl *ImageScanServiceImpl) ConvertEndStepOutputAndSaveVulnerabilities(ste
 		impl.logger.Errorw("error in unmarshalling rendered template", "err", err)
 		return err
 	}
-	allCves := make([]*repository.CveStore, 0, len(vulnerabilities))
+	allCvesMap := make([]*repository.CveStore, 0, len(vulnerabilities))
 	cvesToBeSaved := make([]*repository.CveStore, 0, len(vulnerabilities))
+	uniqueVulnerabilityMap := make(map[string]*bean.ImageScanOutputObject)
+	allCvesNames := make([]string, 0, len(vulnerabilities))
 	for _, vul := range vulnerabilities {
-		cve, err := impl.cveStoreRepository.FindByName(vul.Name)
-		if err != nil && err != pg.ErrNoRows {
-			impl.logger.Errorw("error in getting cve by name", "err", err, "name", vul.Name)
-			return err
+		if _, value := uniqueVulnerabilityMap[vul.Name]; !value {
+			uniqueVulnerabilityMap[vul.Name] = vul
+			allCvesNames = append(allCvesNames, vul.Name)
+		}
+	}
+	allSavedCves, err := impl.cveStoreRepository.FindByCveNames(allCvesNames)
+	if err != nil {
+		impl.logger.Errorw("error in getting all cves ", "err", err)
```

Comment:
should we return?

Reply:
have not returned because of err no rows, first time scanning